framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -175,6 +187,115 @@ private void updateInferredExecutableParameterTypes(
         }
     }
 
+    @Override
+    public void updateContracts(
+            Analysis.BeforeOrAfter preOrPost,
+            ExecutableElement methodElt,
+            AnnotatedTypeFactory atf,
+            CFAbstractStore<?, ?> store) {
+
+        if (store == null) {
+            throw new BugInCF(
+                    "updateContracts(%s, %s, %s, null)",
+                    preOrPost, methodElt, atf.getClass().getSimpleName());
+        }
+
+        String className = getEnclosingClassName(methodElt);
+        String jaifPath = storage.getJaifPath(className);
+        AClass clazz =
+                storage.getAClass(className, jaifPath, ((MethodSymbol) methodElt).enclClass());
+        AMethod amethod = clazz.methods.getVivify(JVMNames.getJVMMethodSignature(methodElt));
+        amethod.setFieldsFromMethodElement(methodElt);
+
+        // TODO: Probably move some part of this into the AnnotatedTypeFactory.
+
+        // TODO: Handle more than just fields of "this".
+
+        // Process the store's fields.
+        for (Map.Entry<FieldAccess, ? extends CFAbstractValue<?>> entry :
+                store.getFieldValues().entrySet()) {
+            FieldAccess fa = entry.getKey();
+            CFAbstractValue<?> v = entry.getValue();
+            VariableElement fieldElement = fa.getField();
+            AnnotatedTypeMirror fieldType = atf.getAnnotatedType(fieldElement);
+
+            AnnotatedTypeMirror atm = convertCFAbstractValueToAnnotatedTypeMirror(v, atf);
+            ((GenericAnnotatedTypeFactory) atf).wpiAdjustForUpdateNonField(atm);
+
+            AField afield = vivifyAndAddTypeMirrorToContract(amethod, preOrPost, fieldElement);
+            storage.updateAnnotationSetInScene(
+                    afield.type, TypeUseLocation.FIELD, atm, fieldType, atf, jaifPath);
+        }
+
+        // Process fields that are not in the store.
+        TypeElement containingClass = (TypeElement) methodElt.getEnclosingElement();
+        ThisReference thisReference = new ThisReference(containingClass.asType());
+        ClassName classNameReceiver = new ClassName(containingClass.asType());
+        for (Element member : containingClass.getEnclosedElements()) {
+            if (member instanceof VariableElement) {
```

Comment:
Instead of `instanceof` you can use `ElementFilter.fieldsIn(containingClass.getEnclosedElements()` as the expression in the foreach.